### ISSUE TITLE
Graph format: graph root 

### ISSUE THREAD

**Issue by dueni0** [SemanticMediaWiki/SemanticResultFormats#543]
### Setup

- MW version: 1.31.1
- SMW version: 3.1.0
- SRF version: 3.1.0

### Issue
Is there a way to start a graph with {{PAGENAME}} or define the graph root ?
In my example pages in Category Person have a property "has_child" and should show all children in a graph. 

Example:
Page: Charles,_Prince_of_Wales

Charles,_Prince_of_Wales 
 ->William, Duke of Cambridge
     -> George Alexander Louis
     -> Charlotte Elizabeth Diana of Cambridge
     ->  Louis Arthur Charles
 -> Harry, Duke of Sussex
     ->  Archie Harrison Mountbatten-Windsor

Page: Harry, Duke of Sussex

Harry, Duke of Sussex
     ->  Archie Harrison Mountbatten-Windsor

**Comment by SbstnS** [comment]
@dueni0: Yes the graph format was designed exactly for such representations!
You should get the expected result by using the following query.

> {{#ask:[[Category:Person]]
>   |?has_child
>   |format=graph
>   |graphcolor=yes
>   |graphlink=yes
>   |graphname="family tree"
>   |graphlabel=yes
>   |graphsize=10,100
>   |nodelabel=displaytitle
> }}

Just let me know if this had worked for you!

**Comment by dueni0** [comment]
@SbstnS: Thank you for your suggestion. At the moment I use a similar query with an additional Category (Category:House_of_Windsor) to filter out other persons. But I would like to start the tree with a certain pagename. It always starts with the "forefather" in this example.

**Comment by gesinn-it** [comment]
It's always good to check the results of a query with format=table to see, if all expected nodes are part of the result.

**Comment by SbstnS** [comment]
If you would like to start the graph with a certain {{FULLPAGENAME}} you could try the following one:

> {{#ask:[[Category:Person]][[has_child::{{FULLPAGENAME}}]]
> |?has_child
> |format=graph
> |graphcolor=yes
> |graphlink=yes
> |graphname="family tree"
> |graphlabel=yes
> |graphsize=10,100
> |nodelabel=displaytitle
> |limit=1000
> }}

I hope this helps you!

**Comment by gesinn-it-gea** [comment]
Closing as answered — @SbstnS's query above (filtering the subject to a specific `{{FULLPAGENAME}}`, e.g. `[[Category:Person]][[has_child::{{FULLPAGENAME}}]]`) is the correct way to root the graph at a specific page; the `graph` format itself has no separate "root" concept, it just renders whatever the query returns. As also noted, `format=table` is a good way to double-check which nodes/edges a query produces before switching to `format=graph`.

Feel free to reopen if this doesn't cover your use case.
